src/ast/Polygon.js

```diff
@@ -49,6 +49,20 @@ class Polygon {
         }
     }
 
+    nameCheck() {
+        this.latlons.forEach((latlon) => {
+            if (isNaN(latlon) && !VarStore.containsName(latlon)) {
```

Comment:
latlon can either be a string or array. In the case of array, I think this might not work...?

Reply:
Is it not always an array in this case?? I'm looping over the array with a for each?